### PR TITLE
Convert docs to quarto and update

### PR DESCRIPTION
**Pull request by jp-dark**
To view the output from these changes:
1. Install required packages:
   ```bash
   python3 -m pip install tiledb-cf[docs]
   ```
2. Make the HTML document:
   ```bash
   make -C docs/ html
   ```
3. Open [docs/_build/html/index.html](./docs/_build/html/index.html) in a web browser of your choice.

### CODE REVIEW

**Comment by jp-dark** [comment]
[sc-33309] [sc-11166]

File: .github/workflows/quarto-render.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - name: "Deploy to gh-pages"
+      uses: peaceiris/actions-gh-pages@v3
+      # Change to the name of your repo's primary branch name:
+      if: github.ref == 'refs/heads/dev'
+      with:
+        # This is GitHub Actions magic; no secrets for us to manage; and this works first-time
+        # without any extra configs other than visiting Settings -> Pages in your GitHub repo.
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: docs
+        destination_dir: docs
```
Comment (perrygeo) [inline .github/workflows/quarto-render.yml:46]: Cool, this quarto+ghactions combo seems like a smooth process! The `peaceiris/actions-gh-pages` action looks legit (I'm slightly paranoid about malicious gh actions so I always check)

File: README.md [hunk 2]
```diff
@@ -25,52 +25,13 @@ pip install tiledb-cf
 
 ### Documentation
 
-#### API Documentation
+Coming soon.
```
Comment (perrygeo) [inline README.md:28]: suggestion: Link to the eventual public URL

File: README.md [hunk 3]
```diff
@@ -25,52 +25,13 @@ pip install tiledb-cf
...
-  --dim-dtype [int8|int16|int32|int64|uint8|uint16|uint32|uint64]
-                                  The data type for TileDB dimensions created from converted NetCDF.  [default: uint64]
-
-  --help                          Show this message and exit.
-```
 
 ## Development
 
-For information on contributing to this project see the [CONTRIBUTING](CONTRIBUTING.md) document.
+For information on contributing to this project see the [contributing](documentation/contributing.md) document and the [code of conduct](documentation/code-of-conduct.md).
```
Comment (perrygeo) [inline README.md:37]: +1

File: _quarto.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+format:
+  html:
+    toc: true
+    theme:
+      light: [flatly, "quarto-materials/tiledb.scss"]
+    mainfont: Helvetica
+    fontsize: 1rem
+    linkcolor: "#4d9fff"
+    code-copy: true
+    code-overflow: wrap
```
Comment (perrygeo) [inline _quarto.yml]: Suggestion: Code wrapping makes several of the code blocks unreadable.

```suggestion
    code-overflow: scroll
```

File: documentation/index.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+To install tiledb-cf with additional dependencies use:
+
+```bash
+pip install tiledb-cf[<optional dependecies>]
+```
+
+For example, to install TileDB-CF-Py and enable the xarray engine with dask support:
+
+```bash
+pip install tiledb-cf[xarray,parallel]
```
Comment (perrygeo) [inline documentation/index.md]: Some shells (zsh in my case) interpret `[]` as shell substitution and needs to be quoted. Most users with esoteric shells would already know this but it doesn't hurt.

 ```suggestion
pip install 'tiledb-cf[xarray,parallel]'
```

File: documentation/tiledb-cf-spec.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+A simple CF dataspace is a direct implementation of the NetCDF data model in TileDB. It follows the same rules as a CF dataspace along with the following requirements:
+
+**Additional Requirements for Dimensions**
+
+1. All dimensions use integer indices and have a domain with lower bound of 0.
+2. All collections of dimensions must have a unique dataspace name.
+
+
+## Specification Q&A
```
Comment (perrygeo) [inline documentation/tiledb-cf-spec.md:48]: +1

File: documentation/xarray-engine.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
+---
+title: TileDB-CF Xarray Engine
+---
+
+
+## Reading from TileDB with Xarray
+
+Xarray uses a plugin infrastructure that allows third-party developers to create their own backend engines for reading and writing. TileDB-CF contains one such backend. To use the backend, make sure `tiledb-cf` is installed in your current Python environment, and use the `tiledb` engine:
```
Comment (perrygeo) [inline documentation/xarray-engine.md]: The xarray plugin architecture is for reading only, right?

File: _quarto.yml [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                - href: "examples/core/group-basics.ipynb"
+
+        - section: "NetCDF Engine"
+          contents:
+            - href: "documentation/netcdf-engine.md"
+            - section: "Examples"
+              contents:
+                - href: "examples/netcdf_engine/netcdf-to-tiledb-basics.ipynb"
+                - href: "examples/netcdf_engine/netcdf-to-tiledb-set-max-fragment-size.ipynb"
+                - href: "examples/netcdf_engine/netcdf-to-tiledb-set-tiles.ipynb"
```
Comment (perrygeo) [inline _quarto.yml:108]: Do these notebooks get run in CI? Or does quarto just render what's in the notebook verbatim?

**Review by perrygeo — APPROVED** [review]
This looks great. The setup process and output of the quarto docs are nicer than sphinx IMO and it's nice to have notebooks included as first-class docs.

I read through the content but tried to resist going down the rabbit hole of going into full "editor" mode. This is very complete and well organized - everything else we can amend as we go.

A few questions and suggestions, nothing blocking.

File: _quarto.yml (hunk 8 above)
Comment (jp-dark) [reply _quarto.yml:108]: It just renders verbatim.

File: documentation/xarray-engine.md (hunk 7 above)
Comment (jp-dark) [reply documentation/xarray-engine.md]: Opps. Correct.

**Pull request merged**